### PR TITLE
Create OFC tests runner script with OFC tests expected results

### PR DESCRIPTION
**Pull request by lukaszkolodziejczyk**

### CODE REVIEW

File: ofc-tests-runner/run_ofc_tests.sh [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+      s)
+        OFC_TESTS_DIR="$OPTARG"
+        ;;
+      o)
+        OFC_OUTPUT_DIR="$OPTARG"
+        ;;
+    esac
+  done
+
+  if [ -z "$OFC_TESTS_DIR" -o ! -d "$OFC_TESTS_DIR" ]; then
```
Comment (franred) [inline ofc-tests-runner/run_ofc_tests.sh]: I bash you can do if [[ -z blah ]] || [[ -d  foo ]] -- but yours might work in non bash shells.

File: ofc-tests-runner/run_ofc_tests.sh [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+  ACTUAL_RESULTS_PROGRAMS_NAME="programs_actual_results"
+  ACTUAL_RESULTS_PROGRAMS_PATH="$OFC_OUTPUT_PROGRAMS_DIR/$ACTUAL_RESULTS_PROGRAMS_NAME"
+  ACTUAL_RESULTS_NIST_NAME="nist_actual_results"
+  ACTUAL_RESULTS_NIST_PATH="$OFC_OUTPUT_NIST_DIR/$ACTUAL_RESULTS_NIST_NAME"
+
+  SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+  EXPECTED_RESULTS_DIR="$SCRIPT_DIR/expected_test_results"
+  EXPECTED_RESULTS_PROGRAMS_PATH="$EXPECTED_RESULTS_DIR/programs_expected_results"
+  EXPECTED_RESULTS_NIST_PATH="$EXPECTED_RESULTS_DIR/nist_expected_results"
+
```
Comment (franred) [inline ofc-tests-runner/run_ofc_tests.sh]: I prefer to write the names of the variables which are part of the script and not in ENV in lower case and if they are not going to be out of this function declare them local, like in the other functions.

File: ofc-tests-runner/expected_test_results/nist_expected_results [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+FM831.FOR:pass
+FM832.FOR:pass
+FM833.FOR:pass
+FM834.FOR:pass
+FM900.FOR:pass
+FM901.FOR:pass
+FM903.FOR:pass
+FM905.FOR:pass
+FM906.FOR:pass
+FM907.FOR:fail
```
Comment (franred) [inline ofc-tests-runner/expected_test_results/nist_expected_results:179]: I though we passed all the nists test

Comment (lukaszkolodziejczyk) [reply ofc-tests-runner/expected_test_results/nist_expected_results:179]: This test also fails when compiled&executed with OFC.
When it executes, the output says the formatting can vary depending on some circumstances. 
This test has .expected file provided in OFC tests with expected output, which is exactly the same as our's (and OFC's), but the formatting is slightly different.
To make it 'pass' here, the .expected file in OFC tests would need to be replaced with the one with our/OFC's formatting.

File: ofc-tests-runner/run_ofc_tests.sh (hunk 1 above)
Comment (lukaszkolodziejczyk) [reply ofc-tests-runner/run_ofc_tests.sh]: Okay, nice to know ! I'm gonna stay with what's there though to keep the convention of 'if's in the script.

File: ofc-tests-runner/expected_test_results/nist_expected_results (hunk 3 above)
Comment (franred) [reply ofc-tests-runner/expected_test_results/nist_expected_results:179]: Ok, maybe check why this fails, I though he said that all the nist tests passed in OFC

**Pull request merged**